### DISCUSSION TITLE
Onlineversion API

### DISCUSSION THREAD

**Discussion by Mephisto090** [arendst/Tasmota#19824] [Q&A] [answered]
Hallo,

für mein ioBroker (jarvis Benutzeroberfläche) würde ich gerne die aktuelle Onlineversion von Tasmota auslesen und im iobroker vergleich um zu wissen, wann ggf. ein Update nötig ist/wäre.

Gibt es irgendwo eine Api wo ich Anfragen kann welche die aktuelle Online Version ist?

Viele Grüße
Florian

---

Hello,

for my ioBroker (jarvis Userinterface) I would like to read out the current online version of Tasmota and use the iobroker comparison to know when an update might be necessary.

Is there an API somewhere where I can query which is the current online version?

Best regards
Florian

**Comment by Jason2866** [comment, marked as answer]
Auto update is a bad idea. There are now and than changes which can break your currently running setup.
Anyways, do you want to look after the release Version or the latest development version?
For the release Version you can use github API to query the actual version https://docs.github.com/en/free-pro-team@latest/rest/releases/releases?apiVersion=2022-11-28#list-releases

**Reply by Mephisto090** [reply]
I don't want to do an auto update. I just want to display it so I can make updates myself

**Reply by sfromis** [reply]
A newer version existing does not indicate that "an update might be necessary". The general recommendation is "if it ain't broke, don't fix it", and better check if each device still works as expected. Of course, if you have issues or want to work based on current docs, that can be a good reason to upgrade.

**Reply by Mephisto090** [reply]
Small example:
![image](https://github.com/arendst/Tasmota/assets/85995176/ad525b4c-3923-41e9-a0c4-076ba4f598aa)

If the version is current it should be displayed as "green" and if there is a more recent version it should be displayed as "red".(13.2.0(tasmota))

**Reply by barbudor** [reply]
You can also read the current release vesion from tasmota_version.h : https://raw.githubusercontent.com/arendst/Tasmota/master/tasmota/include/tasmota_version.h
With a little bit of parsing you should be able to extract it:
```
const uint32_t TASMOTA_VERSION = 0x0D020000;   // 13.2.0.0
```
Just in case, don't trust the comment but parse the constant
